### DISCUSSION TITLE
v2.16

### DISCUSSION THREAD

**Discussion by Ralim** [Ralim/IronOS#1155] [General]
Lots of good changes in here compared to v2.15.

Please note that compared to v2.15, Pinecil files are now `.dfu` files to pack in the device settings _into_ the file to make flashing easier.
The last few `dfu-util` releases support these and they should _just work_ without you needing to specify the address etc.
If you run into issues, make sure you are using the latest `dfu-util`. (Hex files are still provided too).

## Key changes

* Overhaul of the Timer+ADC setup with help from @sandmanRO
* Overhaul of the PID with help from @sandmanRO
* Settings _should_ now upgrade in place to future versions, with resets only happening to new/changed settings
* Shows error if tip runaway (failed temperature sensor) is detected
* USB-PD now has a timeout, to allow forcing QC3 negotiation to start faster
* QC3 Voltages are now adjustable to user desired setpoint
* Added a small tolerance to allow "overvoltage" on QC3 above unit specifications.
* * Please note: Doing this is entirely at your own risk!
* New Advanced view that is much nicer to use and a very good daily driver option from @Mel-kior
* OLED brightness and contrast thanks to @alvinhochun
* Scrollbar is fixed so it doesnt jump around when menus are shown/hidden
* Moved to `.dfu` files from `.bin` to make flashing commands easier for the Pinecil
* Romanian language added
* Rewrite of I2C for Pinecil to resolve issues with the SCA accelerometer locking up I2C
* Move to new PD stack as remaining issues were solved via ^
* Translation updates ❤️ 
* Updated graphics for Idle screen / missing tip
* Updated increments for screen brightness to make more sense

## What's Changed
* Update translation_DE.json by @discip in https://github.com/Ralim/IronOS/pull/961
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/970
* [Docs] Fix how to exit VIN calib by @greenbigfrog in https://github.com/Ralim/IronOS/pull/971
* Update translation_EN.json by @discip in https://github.com/Ralim/IronOS/pull/966
* Fix typo in line 9 of Flashing.md by @marius137 in https://github.com/Ralim/IronOS/pull/968
* Update translation_DE.json by @discip in https://github.com/Ralim/IronOS/pull/972
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/973
* languages (certain unifying) by @discip in https://github.com/Ralim/IronOS/pull/974
* Update translation_UK.json by @playday3008 in https://github.com/Ralim/IronOS/pull/977
* chore(deps): bump actions/cache from 2.1.5 to 2.1.6 by @dependabot in https://github.com/Ralim/IronOS/pull/986
* Update DebugMenu.md by @discip in https://github.com/Ralim/IronOS/pull/981
* Add missing SV translations and fix errors by @piksel in https://github.com/Ralim/IronOS/pull/982
* Update translation_SV.json by @discip in https://github.com/Ralim/IronOS/pull/987
* Fix ci formatter by @Ralim in https://github.com/Ralim/IronOS/pull/991
* USB PD timeout added as an option for improved QC3 compatibility by @VaZso in https://github.com/Ralim/IronOS/pull/994
* Fix documentation typos by @jhenninger in https://github.com/Ralim/IronOS/pull/999
* Update Flashing.md by @discip in https://github.com/Ralim/IronOS/pull/1002
* Thermal Runaway alert by @Ralim in https://github.com/Ralim/IronOS/pull/993
* Add link to Miniware official website by @RichardBronosky in https://github.com/Ralim/IronOS/pull/1013
* Alternate flash command for Pinecil w/o mass-erase by @ironiridis in https://github.com/Ralim/IronOS/pull/1025
* Rework of advanced view by @Mel-kior in https://github.com/Ralim/IronOS/pull/1028
* Rewriting entire settings handling to allow inplace upgrades by @Ralim in https://github.com/Ralim/IronOS/pull/1040
* Add POC contrast and screen invert settings by @alvinhochun in https://github.com/Ralim/IronOS/pull/1010
* Refactor adc by @Ralim in https://github.com/Ralim/IronOS/pull/1039
* updated advanced menu icon by @discip in https://github.com/Ralim/IronOS/pull/1041
* PID Rework by @Ralim in https://github.com/Ralim/IronOS/pull/1043
* Update PL translation by @KITT3000 in https://github.com/Ralim/IronOS/pull/1049
* changed Power - & Advanced - animation by @discip in https://github.com/Ralim/IronOS/pull/1048
* Make QC3 voltages adjustable by @Ralim in https://github.com/Ralim/IronOS/pull/1046
* Update JP translation by @i4M1k0SU in https://github.com/Ralim/IronOS/pull/1051
* Update translation_DE.json by @discip in https://github.com/Ralim/IronOS/pull/1053
* Update translation_EN.json by @discip in https://github.com/Ralim/IronOS/pull/1052
* Create translation_RO.json by @discip in https://github.com/Ralim/IronOS/pull/1056
* PCB -> ACC by @discip in https://github.com/Ralim/IronOS/pull/1059
* Ralim/scrollbar by @Ralim in https://github.com/Ralim/IronOS/pull/1060
* Generate .dfu files by @Ralim in https://github.com/Ralim/IronOS/pull/1058
* Update GUIThread.cpp by @discip in https://github.com/Ralim/IronOS/pull/1061
* Update SK translation by @gamelaster in https://github.com/Ralim/IronOS/pull/1067
* Updates in IT translation by @federicodilo in https://github.com/Ralim/IronOS/pull/1069
* Update translation_SK.json by @OndroNR in https://github.com/Ralim/IronOS/pull/1071
* Update Chinese translations by @alvinhochun in https://github.com/Ralim/IronOS/pull/1072
* translation_DE & translation_EN update by @discip in https://github.com/Ralim/IronOS/pull/1073
* Ralim/named power source by @Ralim in https://github.com/Ralim/IronOS/pull/1074
* Update DebugMenu.md by @discip in https://github.com/Ralim/IronOS/pull/1063
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/1075
* Added LT translations to new entries. by @Vinigas in https://github.com/Ralim/IronOS/pull/1076
* change increment and values of brightness settings by @discip in https://github.com/Ralim/IronOS/pull/1083
* changed power animation by @discip in https://github.com/Ralim/IronOS/pull/1082
* jpg -> png | fixed value alignment for PD timeout by @discip in https://github.com/Ralim/IronOS/pull/1086
* IT translation - second update by @federicodilo in https://github.com/Ralim/IronOS/pull/1088
* Improve translation_JA_JP.json by @yuk7 in https://github.com/Ralim/IronOS/pull/1090
* Update translation_FR.json by @HanaO00 in https://github.com/Ralim/IronOS/pull/1096
* adjusted max brightness by @discip in https://github.com/Ralim/IronOS/pull/1087
* changed idle screen & missing tip by @discip in https://github.com/Ralim/IronOS/pull/1097
* Fixes for I2C on Pinecil + USB-PD stack by @Ralim in https://github.com/Ralim/IronOS/pull/1099
* Fix Japanese translation by @alvinhochun in https://github.com/Ralim/IronOS/pull/1103
* Fix #1105 by @Ralim in https://github.com/Ralim/IronOS/pull/1106
* fix for disconnected tip icon in lefty mode by @discip in https://github.com/Ralim/IronOS/pull/1108
* add new language (Belarussian) by @lidacity in https://github.com/Ralim/IronOS/pull/1123
* Add Greek font table and GR translation by @Devnol in https://github.com/Ralim/IronOS/pull/1121
* Some improvements for Traditional Chinese and Cantonese by @alvinhochun in https://github.com/Ralim/IronOS/pull/1129
* Add custom multi-language build options and improve build docs by @contactalexliu in https://github.com/Ralim/IronOS/pull/1128
* Improve Simplified Chinese translations by @contactalexliu in https://github.com/Ralim/IronOS/pull/1127
* New translations by @onxblog in https://github.com/Ralim/IronOS/pull/1115
* chore(deps): bump actions/cache from 2.1.6 to 2.1.7 by @dependabot in https://github.com/Ralim/IronOS/pull/1130
* Update translation_UK.json by @playday3008 in https://github.com/Ralim/IronOS/pull/1134
* Address settings bounds checks #1116 by @Ralim in https://github.com/Ralim/IronOS/pull/1117
* disconnected tip pic in detailed idle screen by @discip in https://github.com/Ralim/IronOS/pull/1098
* changed POWER_PULSE since 100 would display as 0.0 by @discip in https://github.com/Ralim/IronOS/pull/1136
* Updated translation of the PD section by @onxblog in https://github.com/Ralim/IronOS/pull/1137
* tackling issue #1140 by @discip in https://github.com/Ralim/IronOS/pull/1143
* updated graphics for idle screen & for missing tip by @discip in https://github.com/Ralim/IronOS/pull/1138
* Generic shellscripts by @slycordinator in https://github.com/Ralim/IronOS/pull/1144
* Handle menu sizing better by @Ralim in https://github.com/Ralim/IronOS/pull/1149
* Update IT_translation by @federicodilo in https://github.com/Ralim/IronOS/pull/1153

## New Contributors
* @greenbigfrog made their first contribution in https://github.com/Ralim/IronOS/pull/971
* @marius137 made their first contribution in https://github.com/Ralim/IronOS/pull/968
* @VaZso made their first contribution in https://github.com/Ralim/IronOS/pull/994
* @jhenninger made their first contribution in https://github.com/Ralim/IronOS/pull/999
* @RichardBronosky made their first contribution in https://github.com/Ralim/IronOS/pull/1013
* @ironiridis made their first contribution in https://github.com/Ralim/IronOS/pull/1025
* @i4M1k0SU made their first contribution in https://github.com/Ralim/IronOS/pull/1051
* @gamelaster made their first contribution in https://github.com/Ralim/IronOS/pull/1067
* @yuk7 made their first contribution in https://github.com/Ralim/IronOS/pull/1090
* @lidacity made their first contribution in https://github.com/Ralim/IronOS/pull/1123
* @Devnol made their first contribution in https://github.com/Ralim/IronOS/pull/1121
* @contactalexliu made their first contribution in https://github.com/Ralim/IronOS/pull/1128
* @slycordinator made their first contribution in https://github.com/Ralim/IronOS/pull/1144

**Full Changelog**: https://github.com/Ralim/IronOS/compare/v2.15...v2.16

<hr /><em>This discussion was created from the release <a href='https://github.com/Ralim/IronOS/releases/tag/v2.16'>v2.16</a>.</em>

**Comment by playday3008** [comment]
👍

**Reply by stranger271** [reply]
does not work on the new TS100. The firmware is updated correctly by RDY, but when you restart the power, either the screen is dark or the image has moved to the side. DFU 3.47

**Comment by PUF52** [comment]
Здравствуйте, в v2.16 всё работает очень хорошо, но есть одно замечание. В настройках турбо режима можно установить температуру 450 градусов, реально температура жала может достичь только 428-430 градусов.

**Reply by Ralim** [reply]
While you can adjust up to 450C, you will be limited to the maximum the unit can measure. Which depends entirely upon the ambient temperature which is why it will cap out at around 400C + ambient temperature.

**Reply by PUF52** [reply]
Спасибо, не совсем понятно, как температура окружающей среды влияет на конечную температуру турборежима. Мне так кажется, это влияние можно учесть в прошивке. А то как то получается несоответствие: выставил одно, а получил другое.
Я долго думал, чего ещё не хватает ts100 и понял - звуков. Т.е. было бы хорошо озвучить тональным сигналом нажатия кнопок и все действия при перемещении по меню и подменю. Звук по достижении заданной температуры, уход в сон и отключение и так далее. В качестве источника звука можно использовать пьезоизлучатели и так далее. Как считаете нужно это?

**Reply by Ralim** [reply]
I dont speak Russian, so using below translation

>Thank you, it's not entirely clear how the ambient temperature affects the final turbo temperature. It seems to me that this influence can be taken into account in the firmware. And then somehow it turns out a discrepancy: put one thing, and got another.

>I thought for a long time what the ts100 was still missing and realized - sounds. Those. it would be nice to sound a tone signal for pressing buttons and all actions when moving through menus and submenus. Sound when the set temperature is reached, going to sleep and turning off, and so on. Piezo emitters and so on can be used as a sound source. Do you think this is necessary? 



In terms of how ambient temperature affects the final temperature; this is the main document on how it works https://github.com/Ralim/IronOS/blob/master/Documentation/Temperature.md

But TL;DR is that we measure the tip as its difference from ambient, then use another sensor to measure ambient and add them together.

In terms of sound, If miniware made an Iron that had them like the MHP30 does, I would definitely be open to supporting them.

**Reply by PUF52** [reply]
Здравствуйте. Если я вас правильно понял, то вас смущает отсутствие у паяльника ts100 компании Miniware аппаратных компонентов для реализции воспроизведения звуков, смею вас заверить, что для нас  радиолюбителей, это не проблема. Самое главное для нас, это что бы вы смогли реализовать эту идею (звук) в новом релизе прошивки и вывели его на свободный порт stm32, всё остальное - дело техники. Я имею ввиду электронный ключ, и звуковой излучатель. Спасибо.

**Comment by AlexeyMatveev686** [comment]
Здравствуйте, можете помочь, пожалуйста! Столкнулся с проблемой, что последняя прошивка, с которой стартует мой экземпляр pinecil это 12.14.1. Если ставлю версию новее, то он просто не запускается (пробовал разные версии языков и из 12.15 и из 12.16). В чем может быть проблема и как можно её попробовать решить? Заранее спасибо)

**Comment by paulfertser** [comment]
Hello Alexey,

On Sun, Jan 16, 2022 at 03:23:03AM -0800, Alexey Matveev wrote:
> Здравствуйте, можете помочь, пожалуйста!

Please use English for all your messages in the future. The majority
of the people involved in this project can't read Russian. Thank you.

Regarding your specific issue, please provide all the contents that
the debug menu shows for you in the last version known to work.

If you have already disassembled the iron and took photos, it might be
useful to see PCB markings to see what hardware revision you got.

**Reply by AlexeyMatveev686** [reply]
![20220117_112547](https://user-images.githubusercontent.com/34643750/149740228-ecc955a4-bc7e-4889-a1c6-ee8b10294068.jpg)
![20220117_113052](https://user-images.githubusercontent.com/34643750/149740230-1abd4c13-caff-44b5-bbc3-24111d934477.jpg)
![20220117_113118](https://user-images.githubusercontent.com/34643750/149740231-93c8824f-1190-4f1c-841b-43443b964411.jpg)
![20220117_113138](https://user-images.githubusercontent.com/34643750/149740232-9b1fc68a-1148-49eb-9fed-b5ddea4d45c8.jpg)
![20220117_113212](https://user-images.githubusercontent.com/34643750/149740234-3331fbc1-f65b-4666-b302-cf507127a80a.jpg)
![20220117_113250](https://user-images.githubusercontent.com/34643750/149740236-7cb45eb9-49c4-4c99-b72f-d91ed31999dc.jpg)
![20220117_113313](https://user-images.githubusercontent.com/34643750/149740239-5c1b9486-870a-4990-ad4a-6bcf899b8240.jpg)
![20220117_113336](https://user-images.githubusercontent.com/34643750/149740242-261183af-e5f0-4679-9314-13b31dc03c79.jpg)
![20220117_113342](https://user-images.githubusercontent.com/34643750/149740245-b744e862-9a6b-4021-a08a-0f56a600ab72.jpg)

Good. Sorry for my Russian. This is a photo of my device and a photo of the latest working version I can find

**Reply by discip** [reply]
Good afternoon @AlexeyMatveev686,
by
> all the contents

was meant not only the 1st screen of the debug menu.
If you enter the debug menu and press the *'**+** button'* several times, you will find more information. 😊

**Reply by AlexeyMatveev686** [reply]
![20220118_125843](https://user-images.githubusercontent.com/34643750/149916027-883bb34b-2c7f-4e5c-b49b-2bf13aa15180.jpg)
![20220118_125900](https://user-images.githubusercontent.com/34643750/149916033-ea1e959a-6ab9-44ee-8669-f5349fc5f7a0.jpg)
![20220118_125907](https://user-images.githubusercontent.com/34643750/149916034-35862bfb-9634-4f60-932a-ff4e5b04d92d.jpg)
![20220118_125941](https://user-images.githubusercontent.com/34643750/149916035-ed9e1715-ed8b-47a8-9510-49d4be97de6e.jpg)
![20220118_125956](https://user-images.githubusercontent.com/34643750/149916038-064a3fde-65c3-41ec-9f16-7e50bcd4d43e.jpg)
![20220118_130008](https://user-images.githubusercontent.com/34643750/149916040-2a3ce08b-96e9-478a-b5dc-0ee7217a8096.jpg)
![20220118_130015](https://user-images.githubusercontent.com/34643750/149916044-0b477636-e72d-4b13-8572-89d9c05be4d8.jpg)
![20220118_130024](https://user-images.githubusercontent.com/34643750/149916046-d562745f-6ba9-4d85-8747-44e49c5f7785.jpg)
![20220118_130030](https://user-images.githubusercontent.com/34643750/149916050-6e54c236-b0a1-4e52-ba2d-969d710f6dc5.jpg)
![20220118_130110](https://user-images.githubusercontent.com/34643750/149916051-0fad55bf-cd13-40f9-bacc-aa99a5f1b8e1.jpg)
![20220118_125826](https://user-images.githubusercontent.com/34643750/149916056-e6f8bf4c-bedf-46fe-bfc1-afbeae3d6a6b.jpg)
![20220118_125837](https://user-images.githubusercontent.com/34643750/149916059-ae98a146-2698-4e6d-bcd0-f2f8f461313b.jpg)


Here is all the information and my macbook power supply.

**Reply by AlexeyMatveev686** [reply]
> Good afternoon @AlexeyMatveev686, by
> 
> > all the contents
> 
> was meant not only the 1st screen of the debug menu. If you enter the debug menu and press the _'**+** button'_ several times, you will find more information. 😊

Hello again) Is there any information on my case?

**Reply by discip** [reply]
@AlexeyMatveev686
Good evening,
where did you get your iron from?
If it came from &ensp;***A&nbsp;l&nbsp;i&nbsp;e&nbsp;x&nbsp;p&nbsp;r&nbsp;e&nbsp;s&nbsp;s***,&ensp;it most likely is a clone.

### #1221 seems to be addressing just that.

**Reply by River-Mochi** [reply]
you can join Pinecil discord also, there are community volunteer people is this discord that can help. some speak russian, but usually preferred to use english so more people can help you.   https://discord.gg/Qp78bDdP

also I think you did not get an official pinecil from Pine64.com ?  that looks like one from AliExress ?

**Reply by AlexeyMatveev686** [reply]
@River-b , @discip 
Yes, this one from aliexpress. But I hoped that it will work normally (especially it works fine with older oficial firmware). Thanks for your help anyway. I'll try to find some information in discord.

**Reply by River-Mochi** [reply]
This issue was resolved/closed I believe.
 Alexey/Kobe came to live chat and after some troubleshooting, it ended up being Oled screen contact pins needed to be resoldered.
If other people have this problem, you could add a lot of flux to pins and heat existing solder up to reflow pins to make contact.
just these 2 pins or all the pins in photo. 
There are some youtube videos you can search for repairing TS100 OLED screens, they are similar enough to Pinecil.

![Repaired onea](https://user-images.githubusercontent.com/97197236/153560941-385eecdd-2f33-41f6-b6b7-c25e37255d21.jpg)

**Comment by stranger271** [comment]
does not work on the new TS100. The firmware is updated correctly by RDY, but when you restart the power, either the screen is dark or the image has moved to the side. DFU 3.47 the official firmware at the same time works 2.20

**Reply by VaZso** [reply]
> does not work on the new TS100. The firmware is updated correctly by RDY, but when you restart the power, either the screen is dark or the image has moved to the side. DFU 3.47 the official firmware at the same time works 2.20

About problem with new TS100, please refer to another issue here:
https://github.com/Ralim/IronOS/issues/1177

The problem is in relation of the move from STM32 to GD32 of its microcontroller and a sightly different working of these controllers.

**Reply by stranger271** [reply]
thanks

**Reply by Sch-LikA** [reply]
Apparently same behaviour with a Pinecil. I had a factory 2.14.sth firmware that sometimes hangs when the soldering iron is hot (quite a bad idea). So I upgraded to 2.16, but get a black screen. 2.15 works fine. I tried two times. Staying at 2.15 for now. Is it the same issue ?

Anyway, thanks for the great work guys.

**Reply by discip** [reply]
@Sch-LikA
Please provide enough information, to help the developer to understand what the root for this might be.
1. Where did you buy the device?
2. When did you buy the device?
3. The content of the debug menu. ***Not*** only the version but also information like **`ACC`** ...
4. Maybe you also can provide photos of the board. Since it is easy to disassemble.

**Reply by River-Mochi** [reply]
@Sch-LikA   I believe if it works fine in older firmware but updating to 2.16 is black screen then you probably have bad poor soldering of pins on Oled screen.  This has happened to several other people and is a symtom of poorly soldered OLED screen. See photo I posted above.  The repair is not too hard to do if you are careful and have another soldering iron you could use.  if you can flux/reflow  those OLED pins, this the black screen problem on 2.16 may go away.  however it's hard to know without more information. 
If you want to come to Live community support, people are around and could walk you through some Diagnosis and even the OLED screen re-flux and re-flow solder (carefully peel screen up, add lot of flux, add heat IRon to melt solder again and line up pins).  
OR watch video of people doing TS100 screen repair,  it's similar to Pinecil but there are no videos of Pinecil screen repair that I can find.  you probably don't need to remove whole screen, just add flux and re-heat re-flow the joints with existing solder that is there or add more if needed.

I am personally in Pine64 discord but you can get live community support through any of the Pine64 forum channels.

**Reply by River-Mochi** [reply]
@Sch-LikA  also please open your Pinecil and take detailed photos of front and back of your PCB if you can. but I think there is a good chance your issue can be fixed as long as you have a spare Iron you could use or if you could borrow one.
